pkg/ha/state/lease.go

```diff
@@ -69,44 +75,117 @@ func NewLease(c client.LeaseClient, cluster, potentialLeader string, minT, maxT,
 	}, nil
 }
 
-// UpdateLease uses the supplied client to attempt to update the lease for the potentialLeader.
+func (l *Lease) ValidateSamplesInfo(replica string, minT, maxT, currT time.Time) (bool, time.Time, error) {
+	l._mu.RLock()
+	currentLeader := l.leader
+	currentDeadline := l.leaseUntil
```

Comment:
nit: keep names consistentt
```suggestion
	leaseUntil := l.leaseUntil
```